broker/src/test/java/io/camunda/zeebe/broker/system/configuration/DataCfgTest.java

```diff
@@ -8,25 +8,46 @@
 package io.camunda.zeebe.broker.system.configuration;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
+import io.camunda.zeebe.broker.system.configuration.DiskCfg.FreeSpaceCfg;
+import java.time.Duration;
 import org.junit.Test;
+import org.springframework.util.unit.DataSize;
 
 public class DataCfgTest {
 
   @Test
-  public void shouldSetWatermarksTo1IfDisabled() {
-    // given
+  public void shouldUseDefaultFreeSpaceConfig() {
+    // when
     final DataCfg dataCfg = new DataCfg();
-    dataCfg.setDiskUsageCommandWatermark(0.1);
-    dataCfg.setDiskUsageReplicationWatermark(0.2);
-    dataCfg.setDiskUsageMonitoringEnabled(false);
+    dataCfg.init(new BrokerCfg(), "/base");
 
+    // then
+    assertThat(dataCfg.getDisk().getFreeSpace().getMinFreeSpaceForProcessing("ignore"))
+        .isEqualTo(DataSize.ofGigabytes(2).toBytes());
+    assertThat(dataCfg.getDisk().getFreeSpace().getMinFreeSpaceForReplication("ignore"))
+        .isEqualTo(DataSize.ofGigabytes(1).toBytes());
+  }
+
+  @Test
+  public void shouldOverrideWhenOldWatermarksConfigProvided() {
     // when
+    final DataCfg dataCfg = new DataCfg();
+    final FreeSpaceCfg mockFreeSpaceCfg = mock(FreeSpaceCfg.class);
+    dataCfg.getDisk().setFreeSpace(mockFreeSpaceCfg);
+    dataCfg.setDiskUsageMonitoringInterval(Duration.ofMinutes(5));
+    dataCfg.setDiskUsageMonitoringEnabled(false);
+    dataCfg.setDiskUsageCommandWatermark(0.1);
+    dataCfg.setDiskUsageReplicationWatermark(0.2);
     dataCfg.init(new BrokerCfg(), "/base");
 
     // then
-    assertThat(dataCfg.isDiskUsageMonitoringEnabled()).isFalse();
-    assertThat(dataCfg.getDiskUsageCommandWatermark()).isEqualTo(1.0);
-    assertThat(dataCfg.getDiskUsageReplicationWatermark()).isEqualTo(1.0);
+    verify(mockFreeSpaceCfg, times(1)).setProcessing("90.0%");
```

Comment:
:wrench: IMO we should not mock the `FreeSpaceCfg` and rather check that `dataCfg.getDisk()..getFreeSpace().getMinFreeSpaceForProcessing("ignore")` is as expected. That'd only work if we don't resolve the percentage at startup (see earlier comments from me)

Reply:
I have updated the test to not verify the watermarks. Since it needs to translate percentage to absolute value, it is not possible to check if it assigns the expected value. I guess this is ok.